### PR TITLE
Cloud address translation

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## Usage and product changes

We allow connection to the cloud servers using an address translation mapping (cf. https://github.com/vaticle/typedb-driver/pull/624). This is useful when the route from the user to the servers differs from the route the servers are configured with (e.g. connection to public-facing servers from an internal network).

Example usage: 
```bash
console \
    --cloud=typedb1.domain.com:1729=typedb.local:11729,typedb2.domain.com:1729=typedb.local:21729 \
    --username=<user> --password=<password>
```
or:
```bash
console \
    --cloud=typedb1.domain.com:1729=typedb.local:11729 \
    --cloud=typedb2.domain.com:1729=typedb.local:21729 \
    --username=<user> --password=<password>
```

Note: we currently require that the user provides translation for the addresses of _all_ nodes in the Cloud deployment.

## Implementation

We also update the license headers and file from AGPL to MPL 2.0.

### CODE REVIEW

File: common/Printer.java [hunk 1]
```diff
@@ -96,7 +85,7 @@ private static String stringifyNumericValue(Value value) {
 
     public void databaseReplica(Database.Replica replica) {
         String s = "{ " +
-                colorJsonKey("address: ") + replica.address() + ";" +
+                colorJsonKey(" server: ") + replica.server() + ";" +
```
Comment (dmitrii-ubskii) [inline common/Printer.java:88]: Updated to reflect changes in driver

File: TypeDBConsole.java [hunk 2]
```diff
@@ -498,28 +488,33 @@ private boolean runInlineCommandMode(CLIOptions options, List<String> inlineComm
     }
 
     private TypeDBDriver createTypeDBDriver(CLIOptions options) {
-        TypeDBDriver driver = null;
         try {
+            TypeDBDriver driver;
```
Comment (dmitrii-ubskii) [inline TypeDBConsole.java:492]: In this version, `driver` is never `null`

File: TypeDBConsole.java [hunk 3]
```diff
@@ -498,28 +488,33 @@ private boolean runInlineCommandMode(CLIOptions options, List<String> inlineComm
...
+                }
+            } else {
+                driver = TypeDB.coreDriver(TypeDB.DEFAULT_ADDRESS);
             }
+            return driver;
         } catch (TypeDBDriverException e) {
             printer.error(e.getMessage());
             System.exit(1);
+            return null; // unreachable, but needed to satisfy the compiler
         }
```
Comment (dmitrii-ubskii) [inline TypeDBConsole.java:517]: Java compiler doesn't understand that `System.exit()` diverges.

File: TypeDBConsole.java [hunk 4]
```diff
@@ -498,28 +488,33 @@ private boolean runInlineCommandMode(CLIOptions options, List<String> inlineComm
...
 
     private TypeDBDriver createTypeDBDriver(CLIOptions options) {
-        TypeDBDriver driver = null;
         try {
+            TypeDBDriver driver;
             if (options.core() != null) {
                 driver = TypeDB.coreDriver(options.core());
-            } else {
-                String optCloud = options.cloud();
-                if (optCloud != null) {
```
Comment (dmitrii-ubskii) [inline TypeDBConsole.java:507]: Flattened

File: TypeDBConsole.java [hunk 5]
```diff
@@ -869,10 +864,11 @@ private static class CLIOptions implements Runnable {
 
         @CommandLine.Option(
                 names = {"--cloud"},
-                description = "TypeDB Cloud address to which Console will connect to"
+                description = "TypeDB Cloud address(es) to which Console will connect to, or Cloud address translation 'configured-url=actual-url'",
+                split = ","
         )
         private @Nullable
-        String cloud;
+        String[] cloud;
```
Comment (dmitrii-ubskii) [inline TypeDBConsole.java:871]: picocli treats `--cloud=1,2,3 --cloud=4` as `[1, 2, 3, 4]` with these settings.

File: TypeDBConsole.java [hunk 6]
```diff
@@ -498,28 +488,33 @@ private boolean runInlineCommandMode(CLIOptions options, List<String> inlineComm
...
-                    Optional<Duration> passwordExpiry = driver.users().get(options.username)
-                            .passwordExpirySeconds().map(Duration::ofSeconds);
-                    if (passwordExpiry.isPresent() && passwordExpiry.get().compareTo(PASSWORD_EXPIRY_WARN) < 0) {
-                        printer.info("Your password will expire within " + (passwordExpiry.get().toHours() + 1) + " hour(s).");
-                    }
+            } else if (options.cloud() != null) {
+                String[] optCloud = options.cloud();
+                if (Arrays.stream(optCloud).anyMatch(address -> address.contains("="))) {
+                    Map<String, String> addressTranslation = Arrays.stream(optCloud).map(address -> address.split("=", 2))
+                            .collect(toUnmodifiableMap(parts -> parts[0], parts -> parts[parts.length - 1])); // --cloud=A=B,C becomes { A: B, C: C }
```
Comment (dmitrii-ubskii) [inline TypeDBConsole.java:499]: `--cloud=typedb.local:11729=localhost:1729,typedb.local:21729` becomes `{"typedb.local:11729": "localhost:1729", "typedb.local:21729": "typedb.local:21729"}`, which seems like a reasonable shorthand.

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**